rules/aip0140/abbreviations.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0140
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/googleapis/api-linter/lint"
+	"github.com/jhump/protoreflect/desc"
+)
+
+// checkAbbreviations checks for each long name in the descriptor's name,
+// and returns problems if one is found that suggests the abbreviation instead.
+func checkAbbreviations(d desc.Descriptor, x func(string) string) (problems []lint.Problem) {
```

Comment:
Consider to give `x` a more informative name, e.g., `formatCase` or a better name.

Reply:
Done.